tests/api2/test_012_api_key.py

```diff
@@ -0,0 +1,135 @@
+#!/usr/bin/env python3
+
+# Author: Eric Turgeon
+# License: BSD
+# Location for tests into REST API of FreeNAS
+
+import contextlib
+import io
+import json
+import os
+import sys
+
+import requests
+
+apifolder = os.getcwd()
+sys.path.append(apifolder)
+from functions import POST, GET, DELETE, PUT, SSH_TEST, wait_on_job
+from auto_config import pool_name, scale, ha, password, user, ip
+
+
+@contextlib.contextmanager
+def api_key(allowlist):
+    results = POST('/api_key/', {'name': "Test API KEY", 'allowlist': allowlist})
+    assert results.status_code == 200, results.text
+    key = results.json()
+
+    try:
+        yield key["key"]
+    finally:
+        results = DELETE(f"/api_key/id/{key['id']}/")
+        assert results.status_code == 200, results.text
+
+
+def test_root_api_key_websocket():
+    with api_key([]) as key:
+        cmd = f"midclt -u ws://{ip}/websocket --api-key {key} call system.info"
+        results = SSH_TEST(cmd, user, password, ip)
+        assert results['result'] is True, str(results['output'])
+        assert 'uptime' in str(results['output'])
```

Comment:
May want to print output.

Reply:
Why? Assert prints both sides of the `in` expression if it fails
```
______________________________________________________________________________________ test_a ______________________________________________________________________________________

    def test_a():
        x = 'aaa'
        y = 'bbb'
>       assert x in y
E       AssertionError: assert 'aaa' in 'bbb'

1.py:4: AssertionError
```

Reply:
Sorry. Didn't think this one through.